host/xtest/regression_1000.c

```diff
@@ -1879,3 +1879,82 @@ static void xtest_tee_test_1023(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1023, xtest_tee_test_1023,
 		"Test ELF initialization (.init_array)");
+
+static void xtest_tee_test_1024(ADBG_Case_t *c)
+{
+	TEEC_Session session = { 0 };
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig = 0;
+	uint8_t *empty_buf = NULL;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&session, &os_test_ta_uuid, NULL,
+			&ret_orig)))
+		return;
+
+	empty_buf = malloc(0);
+	if (!empty_buf) {
+		(void)ADBG_EXPECT_TEEC_SUCCESS(c, TEEC_ERROR_OUT_OF_MEMORY);
+		goto out;
+	}
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT);
+
+	/*
+	 * Check temp buffer parameters with a valid address
+	 * but 0 bytes length
+	 */
+	Do_ADBG_BeginSubCase(c,
+			     "Input/Output MEMREF Buffer NULL - Size 0 bytes");
+
+	op.params[0].tmpref.buffer = empty_buf;
+	op.params[0].tmpref.size = 0;
+
+	op.params[1].tmpref.buffer = NULL;
+	op.params[1].tmpref.size = 0;
+
+	op.params[2].tmpref.buffer = empty_buf;
+	op.params[2].tmpref.size = 0;
+
+	op.params[3].tmpref.buffer = NULL;
+	op.params[3].tmpref.size = 0;
+
+	ADBG_EXPECT(c, TEE_SUCCESS,
+		    TEEC_InvokeCommand(&session, TA_OS_TEST_CMD_PARAMS_TMPMEM,
+				       &op, &ret_orig));
+
+	Do_ADBG_EndSubCase(c, NULL);
+
+	/*
+	 * Check temp buffer parameters with a valid address
+	 * but non 0 bytes length
```

Comment:
From this description, we would expect buffer = NULL at lines 1937 and 1943
and size != 0 at line 1941 and 1947.

Maybe passing a unique parameter would be easier to understand the test.
test param#0 = memref IN / ptr=NULL / sz=0 => expect TEEC_SUCCESS.
test param#0 = memref OUT / ptr=NULL / sz=0 => expect TEEC_SUCCESS.
test param#0 = memref IN / ptr=NULL / sz=1 => expect TEEC_ERROR_BAD_PARAMETERS.
test param#0 = memref OUT / ptr=NULL / sz=1 => expect TEEC_ERROR_BAD_PARAMETERS.

Reply:
The current test implementation is verifying expected behavior. I would not change it here.